.changelog/3626.txt

```diff
@@ -0,0 +1,3 @@
+```release-note:feature
+core: Introduce a new CLI command `project destroy` to delete projects in Waypoint and destroy their associated resources
```

Comment:
This is important enough that you can highlight the changelog item like how Brian did the Pipelines changelog here https://github.com/hashicorp/waypoint/pull/3777/files